datumaro/plugins/icdar_format/extractor.py

```diff
@@ -0,0 +1,173 @@
+# Copyright (C) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import os.path as osp
+from glob import glob
+
+from datumaro.components.extractor import (AnnotationType, Bbox, DatasetItem,
+    Importer, Label, LabelCategories, Points, SourceExtractor)
+from datumaro.plugins.icdar_format.format import IcdarPath, IcdarTask
+
+
+class _IcdarExtractor(SourceExtractor):
+    def __init__(self, path, task):
+        if task is IcdarTask.word_recognition:
+            if not osp.isfile(path):
+                raise Exception("Can't read annotation file '%s'" % path)
+            subset = osp.basename(osp.dirname(path))
+            self._dataset_dir = osp.dirname(osp.dirname(osp.dirname(path)))
+        elif task is IcdarTask.text_localization:
+            if not osp.isdir(path):
+                raise Exception("Can't open folder with annotation files'%s'" % path)
+            subset = osp.basename(path)
+            self._dataset_dir = osp.dirname(osp.dirname(path))
+        self._path = path
+        self._task = task
+        super().__init__(subset=subset)
+
+        self._categories = self._load_categories()
+        self._items = list(self._load_items().values())
+
+    def _load_categories(self):
+        categories = {}
+        path = osp.join(self._dataset_dir, IcdarPath.TASK_DIR[self._task],
+            IcdarPath.VOCABULARY_FILE)
+        labels = []
+        if osp.isfile(path):
+            with open(path, encoding='utf-8') as labels_file:
+                labels = [s.strip() for s in labels_file]
+        else:
+            if self._task is IcdarTask.word_recognition:
+                paths = [self._path]
+            else:
+                paths = [p for p in glob(osp.join(self._path, '*.txt'),
+                    recursive=True)]
+            for path in paths:
+                with open(path, encoding='utf-8') as f:
+                    for line in f:
+                        objects = line.split()
+                        if len(objects) == 1:
+                            objects = line.split(',')
+
+                        if len(objects) == 9:
```

Comment:
All these variants are valid and used in the dataset? 
9-values format looks like the segmentation task format. 
2-value format should probably produce `Caption` annotations (word recognition task) and have no labels.

They also say that 

> Please note that the escape character (\) is used for double quotes and backslashes.

so we probably need to support this.

Reply:
9-values and 5-values ​​formats are needed for text localization tasks in [Incidental Scene Text](https://rrc.cvc.uab.es/?ch=4&com=tasks) and [Focused Scene Text](https://rrc.cvc.uab.es/?ch=2&com=tasks) (supported in AC).